### PR TITLE
Remove warnings from tests

### PR DESCRIPTION
**Pull request by haneslinger**
#### Any background context you want to provide?
✨ I recommend reviewing this PR one commit at a time ✨ 
It's big but each PR does _one_ thing. 

💥 There's still alot of errors. 😿 Explained in the ticket #3062 💥 

#### What's this PR do?
This  PR removes many of the warnings that appear when running the tests. 

#### How should this be manually tested?
To test, run 
```
python -Wd manage.py test 
```
#### What are the relevant tickets?
#3062 documents this PR commit by commit, and explains the remaining errors

### CODE REVIEW

**Review by macintoshpie — CHANGES_REQUESTED** [review]
Lookin good! Just a few comments/questions

**Comment by macintoshpie** [comment]
One more thing, it looks like we need to update our project settings for postoffice for this version so that Celery is enabled: https://github.com/ui/django-post_office/tree/v3.6.0#integration-with-celery

Specifically:
in common.py
```
POST_OFFICE = {
    'BACKENDS': {
        'default': 'smtp.EmailBackend',
        'post_office_backend': 'django.core.mail.backends.console.EmailBackend',
    },
    'CELERY_ENABLED': True,
}
```

in docker.py
```
POST_OFFICE = {
    'BACKENDS': {
        'default': EMAIL_BACKEND,
        'post_office_backend': EMAIL_BACKEND,
    },
    'CELERY_ENABLED': True,
}
```

File: requirements/base.txt [hunk 1]
```diff
@@ -66,7 +66,7 @@ django-braces==1.14.0
 # Use forked jwt-oauth2 b/c the published version broke when updating to Django 3.x
 # See: https://docs.djangoproject.com/en/dev/releases/3.0/#removed-private-python-2-compatibility-apis
 # jwt-oauth2>=0.1.1
--e git+https://github.com/SEED-platform/jwt-oauth2.git@0490b817b9b74b7363c4b39fc0a175f58a14c967#egg=jwt-oauth2
+-e git+https://github.com/SEED-platform/jwt-oauth2.git@9726367db36b332c84fb1288a7df32196509c06d#egg=jwt-oauth2
```
[the remark this answers is not in the document]
Comment (haneslinger) [reply requirements/base.txt:69]: Pull request [here](https://github.com/macintoshpie/jwt-oauth2/pull/3). I can't request reviewers 🤔

File: seed/data_importer/models.py [hunk 2]
```diff
@@ -702,6 +702,10 @@ def save(self, in_validation=False, *args, **kwargs):
             pass
             # If we're deleting.
 
+    def __del__(self):
+        if hasattr(self, '_local_file'):
+            self._local_file.close()
```
[the remark this answers is not in the document]
Comment (haneslinger) [reply seed/data_importer/models.py:707]: I agree.

File: seed/lib/mcm/matchers.py [hunk 3]
```diff
@@ -72,7 +72,7 @@ def best_match(s, categories, top_n=5):
             (
                 table_name,
                 category,
-                jellyfish.jaro_winkler(
+                jellyfish.jaro_winkler_distance(
```
[the remark this answers is not in the document]
Comment (haneslinger) [reply seed/lib/mcm/matchers.py:75]: You're right. Idk where I got `jaro_winkler_distance` from.

**Comment by haneslinger** [comment]
| One more thing, it looks like we need to update our project settings for postoffice for this version so that Celery is enabled: https://github.com/ui/django-post_office/tree/v3.6.0#integration-with-celery

Great catch!

File: seed/lib/mcm/matchers.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (haneslinger) [reply seed/lib/mcm/matchers.py]: I figured it out, while the changelog totally  says `jaro_winkler` changed to `jaro_winkler_similarity` in 0.8.0, there's a [note in the doc](https://jamesturk.github.io/jellyfish/functions/#jaro-winkler-similarity) that says this change didn't happen til 0.8.1.

Additionally, the [git blame](https://github.com/jamesturk/jellyfish/commit/8f8546fa5d701d0bbc83f72b90ea7a780eb72c7c) show that there was a point in time where the function was named `jaro_winkler_distance`, but it appears to be attributed to v0.8.2 😵 

Long story short, `jaro_winkler_similarity` doesn't work on 0.8.0. I'm going to update it a few bug patches so the version we are using alines with the docs.

File: requirements/test.txt [hunk 4]
```diff
@@ -14,7 +14,7 @@ Faker==0.9.3
 mock==2.0.0
 nose-exclude==0.5.0
 coveralls-merge==0.0.3
-vcrpy==2.0.1
+vcrpy==4.0.0
```
[the remark this answers is not in the document]
Comment (haneslinger) [reply requirements/test.txt:17]: oh, idk, I've never worked with vcr before. @axelstudios, thoughts?

File: requirements/base.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (macintoshpie) [reply requirements/base.txt:69]: Oops, sorry I meant into SEED's fork

File: requirements/test.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (haneslinger) [reply requirements/test.txt:17]: We've looked into it, vrc was sorta broken even before this PR. It's not to be fixed in this PR, it works here. [Here's the ticket](https://github.com/SEED-platform/seed/issues/3095)

**Review by macintoshpie — APPROVED** [review]
LGTM 👍  Looks like we are going to test/fix vcr in a separate issue
